### ISSUE TITLE
[Bug] [a11y] Unable to tap the Undo button in snackbars

### ISSUE THREAD

**Issue by sv-ohorvath** [mozilla-mobile/fenix#5295]
## Steps to reproduce
1. Enable TalkBack.
2. Close a tab, or delete a bookmark, or other action that can be undone.
3. Try to tap undo.
### Expected behavior
The undo button is focused by TalkBack, users can double tap it.
### Actual behavior
The undo button is not focused by TalkBack, users can't tap it quickly enough.
### Device information

* Android device: Sony Xperia Z5 Premium (Android 7.1.1), Pixel (Android 9)
* Fenix version: 2.0-rc2

**Comment by ekager** [comment]
Maybe we should implement a confirmation dialog instead of the undo snackbar for TalkBack users.

**Comment by brampitoyo** [comment]
@ekager What do you think if we show the snackbar permanently for TalkBack users? The snackbar can only be dismissed if they tap/scroll anywhere outside of its area.

This way, TalkBack users have time to undo any decision, but the snackbar is still really easy to dismiss.

But does it make it _too_ easy to dismiss?

**Comment by ekager** [comment]
You _can_ make a snackbar have length indefinite but I'm not sure how easy they are to dismiss in that way. We'd probably have to set some sort of touch listener/interceptor only for TalkBack users. It would definitely get a bit messy, but I suppose also creating dialogs instead of Snackbars at each of these locations would also.

**Comment by ekager** [comment]
I think if I had to choose one I'd prefer to implement a confirmation dialog and take away undo snackbars for TalkBack users since it makes the action much more deliberate

**Comment by brampitoyo** [comment]
@ekager I’ve just checked Material Design’s snackbar guidelines, and it says:

> Defaults to 4 seconds and can be set up to the maximum duration […]
> Maximum duration allowed […] Set to 10 seconds. If longer durations are necessary than this value consider creating a custom UI for your use case.

Sources:
* https://material.io/develop/ios/components/snackbars/api-docs/Classes/MDCSnackbarMessage.html#/c:objc(cs)MDCSnackbarMessage(py)duration
* https://material.io/develop/ios/components/snackbars/api-docs/Constants.html#/c:@MDCSnackbarMessageDurationMax

I agree with your idea of implementing a persistent-by-default confirmation dialogue. The question is, would it be annoying to be shown a blocking confirmation dialogue after every undo-able actions, or is it an expected behaviour for TalkBack users?

**Comment by brampitoyo** [comment]
Hi @ekager, I’d still love your fedback on implementing a confirmation dialogue in place of the snackbar.

I was concerned that these blocking confirmation dialogues would be annoying if they’re shown after every undo-able action. But maybe this is expected on TalkBack mode?

**Comment by ekager** [comment]
Sorry for the delay @brampitoyo ! We already show these confirmation dialogs in other places where we can't undo deletory actions, so I think this pattern would be familiar. I suggest we move forward with the dialog :)

**Comment by brampitoyo** [comment]
@ekager Sounds good. Let’s do the confirmation dialogue, then.

**Comment by ekager** [comment]
Thanks @brampitoyo ! For now I've added the help wanted labels but we can also discuss in triage if we want to pull this into a sprint

**Comment by sv-sdeiac** [comment]
Verified as fixed using the latest version of Firefox Preview Nightly 1/16 build #20160806.

Devices:
- Nokia 6 (Android 7.1.1);
- Motorola Moto G6 (Android 8.0);
- Samsung Galaxy Note 8 (Android 9);
- Google Pixel (Android Q).
